hazelcast/src/main/java/com/hazelcast/nio/serialization/PortableWriter.java

```diff
@@ -105,16 +112,17 @@
      * @param value     int value to be written
      * @throws IOException in case of any exceptional case
      */
-    void writeShort(String fieldName, short value) throws IOException;
+    void writeShort(@Nonnull String fieldName, short value) throws IOException;
 
     /**
      * Writes a Portable.
+     * Use {@link #writeNullPortable(String, int, int)} to write a {@code null} Portable
```

Comment:
This method seems to accept `null`s (at least in `DefaultPortableWriter`). Does it make sense to provide more context on why the user should prefer `writeNullPortable` over this method in the javadoc?